staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go

```diff
@@ -173,19 +174,40 @@ func validateClaimValidationRules(rules []api.ClaimValidationRule, fldPath *fiel
 	var allErrs field.ErrorList
 
 	seenClaims := sets.NewString()
+	seenExpressions := sets.NewString()
 	for i, rule := range rules {
 		fldPath := fldPath.Index(i)
 
-		if len(rule.Claim) == 0 {
-			allErrs = append(allErrs, field.Required(fldPath.Child("claim"), "claim name is required"))
+		if len(rule.Claim) > 0 {
+			if len(rule.Expression) > 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("expression"), rule.Expression, "claim and expression can't both be set"))
+			}
+			if len(rule.Message) > 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("message"), rule.Message, "message can't be set when claim is set"))
+			}
+			if seenClaims.Has(rule.Claim) {
+				allErrs = append(allErrs, field.Duplicate(fldPath.Child("claim"), rule.Claim))
+			}
+			seenClaims.Insert(rule.Claim)
 			continue
 		}
 
-		if seenClaims.Has(rule.Claim) {
-			allErrs = append(allErrs, field.Duplicate(fldPath.Child("claim"), rule.Claim))
+		if len(rule.Expression) > 0 {
+			// TODO(aramase): do we want to allow empty message?
```

Comment:
Just for reference, in other places when cel expressions are used, setting message is optional, and if it's unset, we return the rule.expression within the returned err message back to user.

Reply:
I don't think the message here is user-facing, but would be in the kube log for the admin